commit 4ae7557a4a5543a66b478dd4fe0aa5e4b660c19b (HEAD -> master)
Author: Clayton Craft <clayton@craftyguy.net>
Date:   Mon Oct 26 05:22:18 2020 +0000

    fix pkgconfig finding enchant-2

diff --git a/meson.build b/meson.build
index a72ea5a..a323d64 100644
--- a/meson.build
+++ b/meson.build
@@ -74,7 +74,7 @@ webkit2gtk = dependency('webkit2gtk-4.0', version: '>=' + target_webkit)
 # https://gitlab.gnome.org/GNOME/geary/issues/439
 appstream_glib = dependency('appstream-glib', version: '>=0.7.10')
 cairo = dependency('cairo')
-enchant = dependency('enchant', version: '>=1.6')
+enchant = dependency('enchant-2', version: '>=1.6')
 folks = dependency('folks', version: '>=0.11')
 gck = dependency('gck-1')
 gcr = dependency('gcr-3', version: '>= 3.10.1')
